sum.go

```diff
@@ -31,6 +41,23 @@ func Sum(data []byte, code int, length int) (Multihash, error) {
 		d, err = sumSHA3(data)
 	case DBL_SHA2_256:
 		d = sumSHA256(sumSHA256(data))
+	case BLAKE2B:
+		switch length {
+		case 32:
+			out := blake2b.Sum256(data)
+			d = out[:]
+		case 48:
+			out := blake2b.Sum384(data)
+			d = out[:]
+		case 64:
+			out := blake2b.Sum512(data)
+			d = out[:]
+		default:
+			return nil, fmt.Errorf("unsupported length for blake2b")
+		}
```

Comment:
We could also say that length is always for truncation and as we are using varint just define blake in two/three bytes range. Like 0xb2 01 to 0xb2 40 is blake2s (second byte is output length) and 0xb2 41 to 0xb2 60 is blake2b (second byte - 0x40 is output length).